### DISCUSSION TITLE
Why does "Disable editing value" disable also creation ?

### DISCUSSION THREAD

**Discussion by emregency** [directus/directus#4126] [Archive]
When `disable editing value` is enable for a field, the field is also disabled on `create` . 

What is the reason behind it 🤔 ?

**Comment by rijkvanzanten** [comment]
It effectively makes the whole field readonly in the app at all times. The idea is that you could use it for things that are generated outside of the app (of even outside Directus), like date-created/user-updated etc.

**Reply by emregency** [reply]
So it means that I can still create those fields through an API call (not UI) but not through the UI, right?

**Reply by rijkvanzanten** [reply]
That's correct, it's a visual app-thing only. You can use permissions to control what's accessible outside of this general on-off "readonly"

**Reply by emregency** [reply]
Awesome 🤩 exactly what I wanted!

**Reply by martin-braun** [reply]
I would actually love to have a "readonly after creation" thingy that also works in the API. Sometimes this can be handy if you want to enable manual primary keys (user input), but you don't want existing IDs to be changed afterwards. Obviously Directus won't allow changing the ID as soon as the data is used in any relation, but I get no error whatsoever. I guess primary keys cannot be changed by design.

**Reply by rijkvanzanten** [reply]
Isn't readonly-after-creation effectively create access but no update access?
